packages/back-end/src/services/experimentNotifications.ts

```diff
@@ -0,0 +1,232 @@
+
+type ExperimentWarningNotificationData = IfEqual<
+  ExperimentNotificationFromCode,
+  ExperimentNotification,
+  ExperimentWarningNotificationPayload,
+  never
+>;
+
+const dispatchEvent = async (
+  context: Context,
+  data: ExperimentWarningNotificationData
+) => {
+  const payload: ExperimentWarningNotificationEvent = {
+    event: "experiment.warning",
+    object: "experiment",
+    data,
+    user: {
+      type: "dashboard",
+      id: context.userId,
+      email: context.email,
+      name: context.userName,
+    },
+  };
+
+  const emittedEvent = await createEvent(context.org.id, payload);
+
+  if (!emittedEvent) throw new Error("Error while creating event!");
+
+  new EventNotifier(emittedEvent.id).perform();
+};
+
+const updateWrapper = async ({
+  context,
+  experiment,
+  type,
+  handler,
+}: {
+  context: Context;
+  experiment: ExperimentInterface;
+  type: ExperimentNotification;
+  handler: () => Promise<void>;
+}) => {
+  if (experiment.pastNotifications?.includes(type)) return;
+
+  await handler();
+
+  await updateExperiment({
+    experiment,
+    context,
+    changes: {
+      pastNotifications: [...(experiment.pastNotifications || []), type],
+    },
+  });
+};
+
+export const notifyFailedAutoUpdate = async ({
+  context,
+  experimentId,
+}: {
+  context: Context;
+  experimentId: string;
+}) => {
+  const experiment = await getExperimentById(context, experimentId);
+
+  if (!experiment) throw new Error("Error while fetching experiment!");
+
+  await updateWrapper({
+    context,
+    experiment,
+    type: "auto-update",
+    handler: () =>
+      dispatchEvent(context, {
+        type: "auto-update",
+        success: false,
+        experimentId,
+        experimentName: experiment.name,
+      }),
+  });
+};
+
+export const notifyAutoUpdateSuccess = async ({
+  context,
+  experimentId,
+}: {
+  context: Context;
+  experimentId: string;
+}) => {
+  const experiment = await getExperimentById(context, experimentId);
+
+  if (!experiment) throw new Error("Error while fetching experiment!");
+
+  if (!experiment.pastNotifications?.includes("auto-update")) return;
+
+  await dispatchEvent(context, {
+    type: "auto-update",
+    success: true,
+    experimentId,
+    experimentName: experiment.name,
+  });
+
+  await updateExperiment({
+    experiment,
+    context,
+    changes: {
+      pastNotifications: (experiment.pastNotifications || []).filter(
+        (v) => v !== "auto-update"
+      ),
+    },
+  });
+};
+
+export const MINIMUM_MULTIPLE_EXPOSURES_PERCENT = 0.01;
+
+const notifyMultipleExposures = async ({
+  context,
+  experiment,
+  lastResult,
+  snapshot,
+}: {
+  context: Context;
+  experiment: ExperimentInterface;
+  lastResult: ExperimentReportResultDimension;
+  snapshot: ExperimentSnapshotDocument;
+}) =>
+  updateWrapper({
+    context,
+    experiment,
+    type: "multiple-exposures",
+    handler: async () => {
+      const totalsUsers = lastResult.variations.reduce(
```

Comment:
Should we consider moving `multiExposureCheck` from the front-end into the shared package so the business logic lives in one place and use that method here?